src/StoreApi/Authentication.php

```diff
@@ -20,15 +22,67 @@ public function init() {
 	 * @return \WP_Error|null|bool
 	 */
 	public function check_authentication( $result ) {
-		if ( ! empty( $result ) ) {
+		if ( ! $this->is_request_to_store_api() ) {
 			return $result;
 		}
 
-		if ( $this->is_request_to_store_api() ) {
-			return true;
+		/**
+		 * Filters the Store API rate limit check, which is disabled by default.
+		 *
+		 * This can be used also to disable the rate limit check when testing API endpoints via a REST API client.
+		 *
+		 * @param boolean $enable_rate_limit_check If true, checks will be enabled.
+		 * @return boolean
+		 */
+		if ( apply_filters( 'woocommerce_store_api_enable_rate_limit_check', false ) ) {
+			$action_id = 'store_api_request_';
+
+			$ip_address = self::get_ip_address();
+
+			if ( is_user_logged_in() ) {
+				$action_id .= get_current_user_id();
+			} else {
+				if ( ! $ip_address ) {
```

Comment:
Thanks a lot for reviewing this @opr 😎.

So we need to tie the requests to a unique key, it will either be a hashed version of IP or User ID depending if the user is logged in or not. To defeat this while using the same IP an attacker would have to use a lot of accounts which would take some effort as well as defeating the IP key, hence why we don't care for IP if the user is logged in.